### ISSUE TITLE
`@typescript-eslint/no-unused-vars`: Do not report reserved interfaces

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#348]
### Description

Hi, as per this [RFC](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md), the interface names `$$Props`, `$$Events`, `$$Slots` are reserved and have a special meaning inside a svelte component. Therefore, they probably shouldn't be reported by `@typescript-eslint/no-unused-vars`.

**Comment by ota-meshi** [comment]
Thank you for posting issue.

I think the parser needs to be fixed to make no-unused-vars work.

The RFC hasn't been merged yet, so I think it should be an option as an experimental feature.
https://github.com/sveltejs/rfcs/pull/38

**Comment by devunt** [comment]
For now I'm using following configuration in `.eslintrc.cjs`. This can provide a quick workaround before RFC lands.

```js
        '@typescript-eslint/no-unused-vars': [
          'warn',
          { varsIgnorePattern: '^\\$\\$(Props|Events|Slots)$' },
        ],
```

**Comment by connerdassen** [comment]
What about $$Generic? 
I get '$$Generic' is not defined eslint(no-undef).
This can be fixed by adding `globals: {$$Generic: "readable"}` to my eslint config, but it leads to problems further on because the type becomes "any", leading to a bunch of "any" usage related errors

**Comment by Razunter** [comment]
Not directly related, but how to handle imported stores?

This throws no-unused-vars:
```
import {store} from '@js/stores/globalStores'

$store = 'abc'
```

**Comment by marekdedic** [comment]
I am closing this as it is irrelevant with Svelte 5 - generics are handled differently and slots and events are deprecated, so it probably isn't worth it supporting these...
